### PR TITLE
implement selection of location in portlet

### LINKED ISSUE #15: portlet shows the city selected on the viewlet
the portlet must allow user to select which of the configured cities it will show.

### PR DESCRIPTION
**Pull request by hvelarde**
fixes #15

### CODE REVIEW

File: src/collective/weather/portlets/weather.py [hunk 1]
```diff
@@ -3,46 +3,91 @@
...
 from zope.component import getUtility
+from zope.formlib import form
 from zope.interface import implements
+from zope.schema.interfaces import IVocabularyFactory
 
 
 class IWeatherPortlet(IPortletDataProvider):
-    """A weather portlet.
-    """
+    '''A weather portlet.
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:16]: according to [PEP 257 -- Docstring Conventions](http://www.python.org/dev/peps/pep-0257/#what-is-a-docstring):

```
For consistency, always use """triple double quotes""" around docstrings
```

so, please do not replace **"""** with **'''**.

File: src/collective/weather/portlets/weather.py [hunk 2]
```diff
@@ -3,46 +3,91 @@
...
+        title=_(u'City'),
+        description=_(u'Choose one of the preconfigured cities'),
+        required=True,
+        vocabulary='collective.weather.Cities')
 
 
 class Assignment(base.Assignment):
-    """Portlet assignment.
-    """
+    '''Portlet assignment.
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:34]: PEP 257

File: src/collective/weather/portlets/weather.py [hunk 3]
```diff
@@ -3,46 +3,91 @@
...
+        self.header = header
+        self.city = city
+
     @property
     def title(self):
-        """This property is used to give the title of the portlet in the
-        "manage portlets" screen.
-        """
-        return _(u'Weather portlet')
+        '''This property is used to give the title of the portlet in the
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:48]: PEP 257

File: src/collective/weather/portlets/weather.py [hunk 4]
```diff
@@ -3,46 +3,91 @@
...
+        'manage portlets' screen.
+        '''
+        return self.header
 
 
 class Renderer(base.Renderer):
-    """Portlet renderer.
-    """
+    '''Portlet renderer.
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:55]: PEP 257

File: src/collective/weather/portlets/weather.py [hunk 5]
```diff
@@ -3,46 +3,91 @@
...
+        self.weather_info = None
+        try:
+            weather_utility.update_weather_info(self.current_city['id'])
+            self.weather_info = weather_utility.get_weather_info(self.current_city)
+        except:
+            pass
+
+
+class AddForm(base.AddForm):
+    '''Portlet add form.
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:77]: PEP 257

File: src/collective/weather/portlets/weather.py [hunk 6]
```diff
@@ -3,46 +3,91 @@
...
+    form_fields = form.Fields(IWeatherPortlet)
+
+    label = _(u'Add Weather Portlet')
+
+    def create(self, data):
+        return Assignment(**data)
+
 
+class EditForm(base.EditForm):
+    '''Portlet edit form.
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:89]: PEP 257

File: src/collective/weather/tests/test_weather_portlet.py [hunk 7]
```diff
@@ -46,7 +46,7 @@ def test_interfaces(self):
         self.assertTrue(IPortletAssignment.providedBy(portlet))
         self.assertTrue(IPortletDataProvider.providedBy(portlet.data))
 
-    @unittest.skip('Portlet has no add view... yet')
+    #@unittest.skip('Portlet has no add view... yet')
```
Comment (hvelarde) [inline src/collective/weather/tests/test_weather_portlet.py]: do not comment, just remove

File: src/collective/weather/tests/test_weather_portlet.py [hunk 8]
```diff
@@ -55,12 +55,12 @@ def test_invoke_add_view(self):
...
 
         addview = mapping.restrictedTraverse('+/' + portlet.addview)
-        addview.createAndAdd(data={})
+        addview.createAndAdd(data={'header': u'Weather', 'city': u'Cordoba'})
 
         self.assertEqual(len(mapping), 1)
         self.assertTrue(isinstance(mapping.values()[0], weather.Assignment))
 
-    @unittest.skip('Portlet has no edit view... yet')
+    #@unittest.skip('Portlet has no edit view... yet')
```
Comment (hvelarde) [inline src/collective/weather/tests/test_weather_portlet.py]: do not comment, just remove

File: src/collective/weather/portlets/weather.py [hunk 9]
```diff
@@ -37,12 +59,38 @@ class Renderer(base.Renderer):
 
     def update(self):
         weather_utility = getUtility(IWeatherUtility)
-        self.weather_info = weather_utility.get_weather_info()
+        factory = getUtility(
+            IVocabularyFactory, name='collective.weather.Locations')
+        vocab = factory(self.context)
+        self.current_city = None
+        if self.data.location in vocab:
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py:66]: what happens if the location is not in the vocabulary?

File: src/collective/weather/portlets/weather.py [hunk 10]
```diff
@@ -37,12 +59,38 @@ class Renderer(base.Renderer):
...
+            location = vocab.by_value[self.data.location]
+            self.current_city = {'id': location.value, 'name': location.title}
+
+        self.weather_info = None
+        try:
+            weather_utility.update_weather_info(self.current_city['id'])
+            self.weather_info = weather_utility.get_weather_info(self.current_city)
+        except:
```
Comment (hvelarde) [inline src/collective/weather/portlets/weather.py]: please, avoid the use of **except:**; we must specify which exception are we trying to catch

File: src/collective/weather/portlets/weather.py (hunk 9 above)
Comment (marcosfromero) [inline src/collective/weather/portlets/weather.py:66]: weather_info is not calculated and the portlet is not shown

**Pull request merged**